robyn/router.py

```diff
@@ -124,11 +125,76 @@ def add_route(
         exception_handler: Optional[Callable],
         injected_dependencies: dict,
     ) -> Union[Callable, CoroutineType]:
+        def wrapped_handler(*args, **kwargs):
+            # In the execute functions the request is passed into *args
+            requests = list(filter(lambda a: isinstance(a, Request), args))
+
+            handler_params = signature(handler).parameters
+
+            if not requests or (len(handler_params) == 1 and next(iter(handler_params)) is Request):
```

Comment:
```suggestion
            if not requests or (len(handler_params) == 1 and isinstance(handler_params[0], Request)):
```

Why not something like this?

Reply:
```
ERROR:robyn.server:Error while executing route function for endpoint `/landing`: 
Traceback (most recent call last):
  File "/home/vishnu/PycharmProjects/Robyn/robyn/router.py", line 202, in inner_handler
    wrapped_handler(*args, **kwargs),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vishnu/PycharmProjects/Robyn/robyn/router.py", line 125, in wrapped_handler
    if not requests or (len(handler_params) == 1 and isinstance(handler_params[0], Request) is Request):
                                                                ~~~~~~~~~~~~~~^^^
 KeyError: 0
 ```
 
 `handler_params` has string indices. we need to know the key to get the first item. using `iter` was the only way I could figure out, CMIIW.

Reply:
alright! gg